packages/babel-plugin-proposal-decorators/test/fixtures/2021-12-classes/initializers/output.js

```diff
@@ -0,0 +1,33 @@
+var _initClass, _initClass2;
+
+let _Foo;
+
+class Foo {}
+
+(() => {
+  [_Foo, _initClass] = babelHelpers.applyDecs(Foo, [], [dec]);
+})();
+
+babelHelpers.defineProperty(Foo, "field", 123);
```

Comment:
Is it correct that this uses `Foo` instead of `_Foo`?

Reply:
So technically I don't think so, it should be using `_Foo`. However, I think that it's not really possible to do this, particularly with static private fields. We would have to extract all of the static fields and assign them to `_Foo` instead, but static private fields need to be defined within a class body to be valid.

I think there's a few options here:

1. Keep it as is, and note this as a limitation of the transform. The biggest meaningful difference would be that static public fields would not be own properties of the class, which I don't _think_ is very common behavior to rely on.
2. Extract all static public fields, leave static private fields, and disregard ordering. I think that classes tend to rely on field initialization order more often than own-ness, so I don't think this is the best option.
3. Extract all static public fields, but assign them within static blocks in between private field assignments. Probably the most _accurate_ transform, but also the most work.